src/elements/fields/MultiFileUploadField.jsx

```diff
@@ -18,19 +18,13 @@ function MultiFileUploadField({
     const showIcon = element.icon_url !== '';
     const showLabel = servar.name !== '';
 
-    // Maintain separate lists of raw files and thumbnails data
     // Raw files are useful to bubble up through onChange
-    // Thumbnails data can be evaluated only when we add files (for performance)
-    const [rawFiles, setRawFiles] = useState(initialFiles);
-    const [thumbnailsData, setThumbnailsData] = useState(
-        initialFiles.map(() => ({ filename: '', thumbnail: '' }))
+    // Raw files are then translated into thumbnail data
+    // And we filter out any files that can't be reasonably thumbnailed
+    const [rawFiles, setRawFiles] = useFileData(initialFiles);
+    const thumbnailData = useThumbnailData(rawFiles).filter(
```

Comment:
can we do this filtering before we set the thumbnail state within the useThumbnailData hook?